### PR TITLE
Upgrade deps

### PR DESCRIPTION
**Pull request by hasezoey**
Update all dependencies
- typescript to 3.7.4
- (all nest) to 6.10.13
- mongoose to 5.8.4
- is-class to 0.0.9
- typegoose to 6.2.1

Removed dependencies
- mockgoose
- rimraf (was nowhere used in the package)
- gulp & gulp-typescript (replaced in favor of plain `tsc`)

Remove `mockgoose` for `mongodb-memory-server` because `Mockgoose` got deprecated

Adding file `tsconfig.build.json` for building

Enforce a more consistent code style 

Sorted tslint.json

Building now excludes all tests & `.tsbuildinfo`, this will result in noticeable size decrease

Add "if" in travis to not create a job outside of `npm-release`

Adding `test:watch` & `lint` & `prepare` as scripts to package.json

Made builds (.travis.yml) faster & more checking (by removing the need to run npm-publish on Pull Requests, moving coverage to unit-tests, adding lint stage)

---

This PR will fix https://github.com/kpfromer/nestjs-typegoose/pull/40#issuecomment-552015769

@kpfromer i recommend using a squash merge

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,12 @@ install:
 jobs:
   include:
     - stage: unit tests
-      script: yarn test:ci
+      script: yarn run test:ci
     - stage: build
-      script: yarn build
-      after_success: npm run coverage
+      script: yarn run build
+      after_success: yarn run coverage
```
Comment (hasezoey) [inline .travis.yml]: i would recommend combining the coverage upload with the tests

File: src/typegoose.providers.spec.ts [hunk 2]
```diff
@@ -24,27 +24,31 @@ describe('createTypegooseProviders', () => {
...
-      });
-    }, 60000);
 
+    connection = await mongoose.createConnection(await mongod.getConnectionString(), {
+      useCreateIndex: true,
+      useFindAndModify: true,
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    });
+    // setTimeout(() => {
```
Comment (hasezoey) [inline src/typegoose.providers.spec.ts]: before merging i wanted to ask why this was here, and if it could be removed

**Comment by hasezoey** [comment]
@kpfromer when you have the time, i would recommend switching travis to be a GitHub app (and from travis.org to travis.com)

File: .travis.yml (hunk 1 above)
Comment (kpfromer) [reply .travis.yml]: Yeah, I think that would be good.

File: src/typegoose.providers.spec.ts (hunk 2 above)
Comment (kpfromer) [reply src/typegoose.providers.spec.ts]: I was getting deprecation warnings from mongoose regarding schema options. I followed that they recommended at the time and added these options.

Comment (hasezoey) [reply src/typegoose.providers.spec.ts]: > I followed that they recommended at the time and added these options.

sorry if github was not showing this correctly, but i meant the commented-out `setTimeout` block

Comment (kpfromer) [reply src/typegoose.providers.spec.ts]: Yeah, it can be removed.

Comment (hasezoey) [reply src/typegoose.providers.spec.ts]: removed in 4de39a3ef984669f0d3b65118fb7e2283e2d7fb9

**Comment by hasezoey** [comment]
Note: because this project didnt have an config (/ switch for local mongodb), i cant test it locally anymore (requires libcurl3, but i need libcurl4 for other things), and didnt want to add too much overhead

File: .travis.yml (hunk 1 above)
Comment (hasezoey) [reply .travis.yml]: moved in 006ee969237217d23d95942249deace5d23a0ff7

**Comment by hasezoey** [comment]
Updated Dependencies again to be the latest (typegoose 6.1.1 to 6.1.5, mongoose, and all nestjs)

**Comment by hasezoey** [comment]
i would say this PR is ready to merge, just want to say some things before:
- i would recommend to test these changes on an nestjs-testing bed project to confirm it is working (im not so with nestjs currently)
- i would recommend to check what changed since last review

File: src/typegoose.module.spec.ts [hunk 3]
```diff
@@ -129,7 +129,7 @@ describe('TypegooseModule', () => {
     it('should return module that createdTypegooseProviders with provided connectionName', () => {
       const connectionName = 'OtherMongoDB';
 
-      const module = TypegooseModule.forFeature(models, connectionName);
+      const module = TypegooseModule.forFeature(models, connectionName); // is this wanted?
```
Comment (hasezoey) [inline src/typegoose.module.spec.ts]: is this wanted to be run? when yes, the value could be removed because it is unused

Comment (kpfromer) [reply src/typegoose.module.spec.ts]: It is unnecessary. I think removing it would be best.

**Comment by kpfromer** [comment]
I have also tested it in an example application and it works as intended.

File: src/typegoose.module.spec.ts (hunk 3 above)
Comment (hasezoey) [reply src/typegoose.module.spec.ts]: just to confirm, the whole line, or just the value assignment?

**Comment by hasezoey** [comment]
@kpfromer i upgraded the dependencies again (except nest), to match the latests versions of mongoose & typegoose, as for mongodb-memory-server, it got a major update that replaced getos (which now allows me to fully run the tests because it now works on Linux Mint)
-> i will upgrade nestjs in a seperate commit

**Comment by hasezoey** [comment]
Now upgraded to the latest (minor) nestjs version, merged master to not conflict with the merge into master, and added "yarn run lint" to the building stage

**Pull request merged**